stout/range.h

```diff
@@ -0,0 +1,92 @@
+#pragma once
+
+#include <algorithm>
+
+#include "stout/stream.h"
+
+////////////////////////////////////////////////////////////////////////
+
+namespace stout {
+namespace eventuals {
+
+////////////////////////////////////////////////////////////////////////
+
+namespace detail {
```

Comment:
We're slowing moving away from using the `detail` namespace and just using the prefixed `_` instead, so please drop!